# Современные методы анализа данных и машинного обучения, БИ

## НИУ ВШЭ, 2025-26 учебный год

## Групповой проект №4. Statistics in Data Analysis


## Описание задания

Практика A/B-тестирования, формирования и проверки статистических гипотез лежит в основе принятия решений в любой современной компании, работающей с данными. Умение корректно провести эксперимент, проинтерпретировать результаты и сформулировать выводы непосредственно влияет на возможность роста продукта, сокращение издержек и уменьшение рисков.


Данный групповой проект состоит из **двух частей**:

1. **A/B-тестирование**: анализ результатов эксперимента, которые зафиксированы в датасете `data_ab.csv`, — проверка наличия статистически и практически значимых различий между наблюдениями в контрольной и экспериментальной группах.
2. **Проверка гипотез на выбранном датасете**: выбор одного или нескольких датасетов с сайта [UCI Machine Learning Repository](https://archive.ics.uci.edu/) либо же получение соответствующих датасетов путем парсинга/работы с API; осуществление этапа первичного анализа данных (EDA); определение не менее трёх содержательных гипотез **на основе проделанного EDA**; их проверка с помощью различнных статистических тестов.


### Постановка задачи

#### Часть 1. A/B-тест

Для выполнения первой части группового проекта вам предоставляется датасет с результатами проведения некоторого A/B-тестирования в компании из сферы образования.


Датасет состоит из пяти столбцов:

* `user_id` — уникальный идентификатор пользователя;

* `timestamp` — время посещения пользователем страницы;

* `group` — группа эксперимента, к которой был случайно отнесён пользователь:

  *   `control` — контрольная группа,
  *   `treatment` — экспериментальная группа;

* `landing_page` — версия лендинга, которую увидел пользователь:

  *   `old_page` — старая версия страницы,
  *   `new_page` — новая версия страницы;

* `converted` — бинарный признак, показывающий, совершил ли пользователь целевое действие (целевым действием является регистрация на курс).








По замыслу A/B-теста контрольной группе (`control`) должна была показываться старая версия лендинга — `old_page`; экспериментальной группе (`treatment`) должна была показываться новая версия лендинга — `new_page`.

Вам следует:
- понять структуру данных, провести EDA и выбрать метрику, с которой будете работать;
- проверить корректность проведения тестирования (посмотреть на однородность групп; задуматься об A/A-тесте; не забыть, что в нашем распоряжении есть еще и временные метки);
- сформулировать нулевую и альтернативную гипотезы на основании выбранной метрики;
- выбрать и обосновать статистический тест;
- решить, нужно ли устаналивать значения каких-либо параметров до начала анализа тестирования;
- оценить разницу в результатах и сделать выводы о практической значимости эксперимента;
- провести дополнительный анализ при необходимости (например, с помощью [bootstrap](https://habr.com/ru/companies/X5Tech/articles/679842/)  или через анализ чувствительности);
- сформулировать рекомендации и бизнес-инсайты по результатам эксперимента.

**Важно**: это лишь возможные шаги для вашего исследования, и воспринимать их следует как подсказки. Ваша центральная задача в любом случае представлена в общем виде — осуществить качественный и полный анализ результатов проведенного A/B-тестирования.



#### Понять структуру данных, провести EDA и выбрать метрику, с которой будете работать;

In [12]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [15]:
data = pd.read_csv('data_abс.csv')

In [16]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 294478 entries, 0 to 294477
Data columns (total 5 columns):
 #   Column        Non-Null Count   Dtype 
---  ------        --------------   ----- 
 0   user_id       294478 non-null  int64 
 1   timestamp     294478 non-null  object
 2   group         294478 non-null  object
 3   landing_page  294478 non-null  object
 4   converted     294478 non-null  int64 
dtypes: int64(2), object(3)
memory usage: 11.2+ MB


294478 строчек, и 5 столбцов, нет пропусков. Проверим есть ли дубликаты

In [17]:
data = data.drop_duplicates()
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 294478 entries, 0 to 294477
Data columns (total 5 columns):
 #   Column        Non-Null Count   Dtype 
---  ------        --------------   ----- 
 0   user_id       294478 non-null  int64 
 1   timestamp     294478 non-null  object
 2   group         294478 non-null  object
 3   landing_page  294478 non-null  object
 4   converted     294478 non-null  int64 
dtypes: int64(2), object(3)
memory usage: 11.2+ MB


Дубликатов нет

###проверить корректность проведения тестирования (посмотреть на однородность групп; задуматься об A/A-тесте; не забыть, что в нашем распоряжении есть еще и временные метки);

#### Сформулировать нулевую и альтернативную гипотезы на основании выбранной метрики;

Гипотеза группы завлеченных клиентов в train и test равны

#### Часть 2. Проверка гипотез методами математической статистики

В этой части группового проекта вам предстоит самим выбрать предметную область, а также самим получить датасет или датасеты, с которыми вы будете работать.

С точки зрения источников данных у вас есть две опции: вы можете найти ваш датасет — или 2-3 датасета в случае необходимости — на сайте [UCI Machine Learning Repository](https://archive.ics.uci.edu/) (брать датасеты необходимо именно с этого сайта); либо же вторая опция — можно собрать датасет/датасеты вручную с помощью средств Parsing'a и API. Допускается использование и комбинации этих двух опций. В любом случае, **каждый** используемый вами датасет должен (до этапа EDA) содержать **не менее 2000 строк** и **не менее 8 признаков**.




После получения датасетов вам следует:
- провести EDA с осмысленными визуализациями и качественным поиском первичных закономерностей;
- на основе проделенного EDA сформулировать **минимум три чёткие, содержательные гипотезы**;
- подобрать и реализовать **минимум три различных статистических теста** для проверки сформулированных ранее гипотез, причём каждый тест должен быть обязательно обоснован (различные статистические тесты в нашем случае означает, что статистические критерии для этих тестов должны быть разными);
- корректно и полно проинтерпретировать полученные результаты и сформулировать итоговые выводы.

**Важно:** вы должны очень ответственно подойти к выбору датасета, так как гипотезы должны быть **не случайными**, а логически вытекающими из осуществленного EDA и, кроме того, имеющими реальный интерес для изучения и проверки с точки зрения бизнес-составляющей. В случае если по одному датасету у вас не получается сформировать 3 достаточно осмысленных гипотезы — разрешается использовать суммарно вплоть до 3 датасетов. В таком случае, все эти датасеты должны иметь какую-то связь между собой.

### Общие требования к проекту

В качестве результата выполнения группового проекта ваша команда должна получить презентацию и защитить ее перед комиссией. Наличие репозитория с кодом на GitHub является необязательным условием, но приветствуется.

Оформление презентации остаётся полностью на ваше усмотрение, но помните, что результат должен быть релевантен для демонстрации бизнес-заказчику — комиссию, принимающую вашу работу, правильнее всего воспринимать именно в таком качестве. Например, вставлять в презентацию строчки кода или злоупотреблять скринами блокнота не рекомендуется.

С точки зрения концепции выполнения проекта — вам необходимо принять на себя роль дата-аналитиков: продемонстрировать способность грамотно работать с экспериментальными статистическими данными, что включает в себя проверку корректности проведения и интерпретацию результатов построенного A/B-тестирования, правильную формулировку гипотез, выбор подходящих статистических методов, моделей и их параметров, а также построение итоговых выводов из анализа — и исследователей: уметь выбирать релевантный датасет, проводить содержательный EDA, находить закономерности и превращать их в осмысленные статистические гипотезы для проверки. А в конце, конечно, — презентовать всю эту красоту в понятном виде потенциальному заказчику.

Подчеркнём отдельно: ваша задача — не просто провести тесты и сделать какие-то выводы из них, но разобраться в данных, изучить их природу, выявить реальные зависимости и инсайты, а также на основе проделанного анализа показать, почему эти находки важны для бизнеса или для выбранного контекста датасета. Разумеется, чем интереснее выбранная вами предметная область, чем осмысленнее сформулированные гипотезы и чем глубже будет проведенное вами исследование — тем выше окажется итоговое качество проекта и тем выше он будет оценён!

### Полезные замечания и советы

- Графики и визуализации помогают наглядно доносить информацию, а также анализировать данные
- Формулируйте гипотезы *после* EDA. Не придумывайте гипотезы "вслепую", сначала погрузитесь в данные, найдите закономерности, паттерны и лишь затем переходите к выделению статически проверяемых гипотез
- Не забывайте о прикладной значимости ваших проектов — ваш анализ должен быть полезен потенциальному заказчику
- Используйте разные подходы к проверке гипотез. Ищите способы усилить ваши тесты. Применяйте новые методы и идеи, а затем рассказывайте про них нам. Не бойтесь экспериментировать! Это будет только поощряться :)


## Критерии оценки проекта

Оценка за групповой проект в целом — то есть первая из двух оценок — выставляется по 10 балльной шкале, исходя из 3 критериев. Также комиссия на своё усмотрение может добавить (но не снизить) некоторое количество бонусных баллов за рамками данных критериев.

Каждый из критериев будет перечислен и рассмотрен ниже.

### Критерий 1. Анализ результатов A/B-тестирования (до 3,5 баллов)

Данный критерий оценивает качество, глубину и корректность проведенного анализа результатов A/B-тестирования, включая выбор метрики, оценку реализации теста, а также спектр статистических методов и их обоснованность, формулирование гипотез и выводов по проделанному исследованию.

*По данному критерию можно получить **1 балл и выше** только при условии осмысленного осуществления этапа EDA, корректного выбора метрики, верной формулировки гипотез, обоснованного проведения статистического теста и правильной интерпретации полученных результатов.*

*По данному критерию можно получить **2 балла и выше** только при условии учёта в значительной степени временной составляющей эксперимента.*

*По данному критерию можно получить **3 балла и выше** только при условии обоснованного применения дополнительных инструментов/подходов для анализа результатов A/B-тестирования, не упомянутых ранее в этом ноутбуке и не рассматривавшихся в рамках соответствующего семинара и лекции (bootstrap'a или анализа чувствительности может хватить на 3 балла при условии корректного и осмысленного их применения).*




### Критерий 2. Выбор предметной области и датасета, EDA, формулирование гипотез (до 3 баллов)

Данный критерий оценивает, насколько обоснованными и осмысленными получились сформулированные вами гипотезы на основании проведенного EDA по выбранной вами второй предметной области и датасетам, связанным с ней.

*__Важно:__ в случае оценки __0 баллов__ за этот критерий оценка за следующий критерий также становится равна нулю.*

*По данному критерию нельзя получить **больше 0 баллов** при отсутствии 3 сформулированных по датасету или связанным друг с другом датасетам, гипотез, имеющих **разную природу**.*

*По данному критерию нельзя получить **больше 0 баллов** при условии, что хотя бы один используемый в проекте датасет не удовлетворяет требованиям, описанным в разделах выше.*

*По данному критерию можно получить **1 балл и выше** только при условии осмысленного осуществления этапа EDA и наличия 3 гипотез, напрямую логически вытекающих из этого этапа.*


### Критерий 3. Проведение статистических тестов по выбранной предметной области (до 3,5 баллов)

Данный критерий оценивает качество статистического анализа, корректность выбора тестов, разнообразие использованных статистических инструментов, интерпретацию результатов и глубину полученных выводов.

*По данному критерию можно получить **1 балл и выше** только при условии корректного использования трех различных статистических тестов.*

*По данному критерию можно получить **2 балла и выше** только при условии полностью обоснованного (и правильного) выбора каждого из используемых статистических тестов.*

*По данному критерию можно получить **3 балла и выше** только при условии проведения сравнительного анализа нескольких тестов друг с другом хотя бы для одной гипотезы, а также осмысленной интерпретации проделанного анализа в контексте реальной исследовательской задачи.*